quodlibet/quodlibet/ext/playorder/shufflebygrouping.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Phoenix Dailey
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+import random
+
+from gi.repository import Gtk, GLib
+
+from quodlibet import _
+from quodlibet import app
+from quodlibet.order import OrderInOrder
+from quodlibet.order import OrderRemembered
+from quodlibet.order.reorder import Reorder
+from quodlibet.plugins import PluginConfig
+from quodlibet.plugins.playorder import ShufflePlugin
+from quodlibet.qltk import Icons
+
+
+pconfig = PluginConfig("shufflebygrouping")
+pconfig.defaults.set("grouping", "~grouping~album~albumartist")
+pconfig.defaults.set("grouping_test", "grouping")
+pconfig.defaults.set("delay", 0)
+
+
+class ShuffleByGrouping(ShufflePlugin, OrderInOrder, OrderRemembered):
+    PLUGIN_ID = "shufflebygrouping"
```

Comment:
On a recent count we had four separate naming conventions for this (argh), but the most common is just `Title Case Plugin` (35ish) `followed by `CamelCasePlugin` (17ish), less for `snake_case` and least for `modulestyle` as here, so it'd be great if you went for one of the first two...